src/components/app.jsx

```diff
@@ -39,9 +41,24 @@ const PublicContext = withClient(({ client }) => {
   useEffect(() => {
     const fetchData = async () => {
       try {
-        const { id, readOnly } = await getSharedDocument(client)
-        setReadOnly(readOnly)
-        setSharedDocumentId(id)
+        const searchParams = new URLSearchParams(window.location.search)
+        if (searchParams.has('id')) {
+          // public route = /public/?sharecode=xxxx&id=xxxxx
+          // The id of the note is necessary because the sharecode
+          // may be about a folder and not a specific note.
+          const id = searchParams.get('id')
+          const readOnly = await isNoteReadOnly(client, id)
```

Comment:
Having a `isXXX` method do not me think about having to make an async call. Since we're fetching a few information, `fetchIfNoteIsReadOnly()` or something similar is more understandable